Source/Diagnostics/ReducedDiags/ChargeOnEB.cpp

```diff
@@ -0,0 +1,204 @@
+#include <AMReX_REAL.H>
+
+#include <algorithm>
+#include <fstream>
+#include <vector>
+
+using namespace amrex;
+
+// constructor
+ChargeOnEB::ChargeOnEB (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    // Only 3D is working for now
+#if !(defined WARPX_DIM_3D)
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+        "ChargeOnEB reduced diagnostics only works in 3D");
+#endif
+
+#if !(defined AMREX_USE_EB)
+    WARPX_ALWAYS_ASSERT_WITH_MESSAGE(false,
+        "ChargeOnEB reduced diagnostics only works when compiling with EB support");
+#endif
+
+    // resize data array
+    m_data.resize(1, 0.0_rt);
+
+    // Read optional weighting
+    std::string buf;
+    ParmParse pp_rd_name(rd_name);
+    m_do_parser_weighting = pp_rd_name.query("weighting_function(x,y,z)", buf);
+    if (m_do_parser_weighting) {
+        std::string weighting_string = "";
+        utils::parser::Store_parserString(
+            pp_rd_name,"weighting_function(x,y,z)", weighting_string);
+        m_parser_weighting = std::make_unique<amrex::Parser>(
+            utils::parser::makeParser(weighting_string,{"x","y","z"}));
+    }
+
+    if (ParallelDescriptor::IOProcessor())
+    {
+        if ( m_IsNotRestart )
+        {
+            // open file
+            std::ofstream ofs{m_path + m_rd_name + "." + m_extension, std::ofstream::out};
+            // write header row
+            int c = 0;
+            ofs << "#";
+            ofs << "[" << c++ << "]step()";
+            ofs << m_sep;
+            ofs << "[" << c++ << "]time(s)";
+            ofs << m_sep;
+            ofs << "[" << c++ << "]Charge (C)";
+            ofs << std::endl;
+            // close file
+            ofs.close();
+        }
+    }
+}
+// end constructor
+
+// function that computes the charge at the surface of the EB
+void ChargeOnEB::ComputeDiags (int step)
+{
+    // Judge whether the diags should be done
+    if (!m_intervals.contains(step+1)) { return; }
+
+#if ((defined WARPX_DIM_3D) && (defined AMREX_USE_EB))
+    // get a reference to WarpX instance
+    auto & warpx = WarpX::GetInstance();
+
+    // Only compute the integral on level 0
+    int const lev = 0;
+
+    // get MultiFab data at lev
+    const MultiFab & Ex = warpx.getEfield(lev,0);
+    const MultiFab & Ey = warpx.getEfield(lev,1);
+    const MultiFab & Ez = warpx.getEfield(lev,2);
+
+    // get EB structures
+    amrex::EBFArrayBoxFactory const& eb_box_factory = warpx.fieldEBFactory(lev);
+    amrex::FabArray<amrex::EBCellFlagFab> const& eb_flag = eb_box_factory.getMultiEBCellFlagFab();
+    amrex::MultiCutFab const& eb_bnd_cent = eb_box_factory.getBndryCent();
+    amrex::MultiCutFab const& eb_bnd_normal = eb_box_factory.getBndryNormal();
+    amrex::Array<const amrex::MultiCutFab*,AMREX_SPACEDIM> eb_area_fraction = eb_box_factory.getAreaFrac();
+
+    // get surface integration element
+    const auto& warpx_instance = WarpX::GetInstance();
+    const auto dx = warpx_instance.Geom(lev).CellSizeArray();
```

Comment:
Can we use here the WarpX instance `warpx` already defined above?
```suggestion
    const amrex::GpuArray<amrex::Real,AMREX_SPACEDIM> dx = warpx.Geom(lev).CellSizeArray();
```